library/alloc/src/vec.rs

```diff
@@ -2386,22 +2386,46 @@ where
 impl<T> SpecExtend<T, IntoIter<T>> for Vec<T> {
     fn spec_extend(&mut self, iterator: IntoIter<T>) {
         // Avoid reallocation if we can use iterator's storage instead. This requires 1 memcpy and 0-1 memmove
-        // while reallocation would require 1 alloc, 1-2 memcpy, 1-2 free
+        // while reallocation would require 1 alloc, 1-2 memcpy, 1-2 free.
+        //
+        // ## non-empty self, partially consumed iterator
```

Comment:
Maybe we could mention how many memcpy and memmove for each diagram?

Reply:
But why? Is it not obvious?

Reply:
Obvious but I need to give it some thoughts, I can't even read memcpy or memmove from assembly.

Reply:
Done.